### PR TITLE
Implement an action to manually change the party leader.

### PR DESCRIPTION
**Pull request by 2Pacalypse-**

### CODE REVIEW

**Comment by 2Pacalypse-** [comment]
Depends on https://github.com/ShieldBattery/ShieldBattery/pull/731

File: server/lib/parties/party-service.ts [hunk 1]
```diff
@@ -314,6 +314,41 @@ export default class PartyService {
...
+      throw new PartyServiceError(
+        PartyServiceErrorCode.InsufficientPermissions,
+        'Only party leaders can change leaders',
+      )
+    }
+
+    if (!party.members.has(newLeader.id)) {
+      throw new PartyServiceError(
+        PartyServiceErrorCode.InvalidAction,
+        "Can't make leader a player who is not in your party",
```
Comment (tec27) [inline server/lib/parties/party-service.ts:336]: nit: this is a bit hard to parse, maybe `Only party members can be made leader`

**Review by tec27 — APPROVED** [review]
